diesel/src/pg/connection/mod.rs

```diff
@@ -119,33 +237,26 @@ impl GetPgMetadataCache for PgConnection {
 #[inline(always)]
 fn update_transaction_manager_status<T>(
     query_result: QueryResult<T>,
-    conn: &mut PgConnection,
+    conn: &mut ConnectionAndTransactionManager,
 ) -> QueryResult<T> {
     if let Err(Error::DatabaseError { .. }) = query_result {
         /// avoid monomorphizing for every result type - this part will not be inlined
-        fn non_generic_inner(conn: &mut PgConnection) {
-            if AnsiTransactionManager::transaction_manager_status_mut(conn)
-                .is_not_broken_and_in_transaction()
-            {
+        fn non_generic_inner(raw_conn: &mut RawConnection, tm: &mut AnsiTransactionManager) {
```

Comment:
I've opted for two arguments here as this makes accessing the relevant parts in this function easier in my opinion.

Reply:
As this code is only hit in case of error, I would rather like to optimize for readability instead of performance as performance won't matter that much here.

Reply:
:+1: